src/main/java/com/dev/cinema/controllers/OrderController.java

```diff
@@ -0,0 +1,51 @@
+package com.dev.cinema.controllers;
+
+import com.dev.cinema.model.dto.order.OrderDtoMapper;
+import com.dev.cinema.model.dto.order.OrderResponseDto;
+import com.dev.cinema.service.OrderService;
+import com.dev.cinema.service.ShoppingCartService;
+import com.dev.cinema.service.UserService;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/orders")
+public class OrderController {
+    private final OrderService orderService;
+    private final ShoppingCartService shoppingCartService;
+    private final UserService userService;
+    private final OrderDtoMapper orderDtoMapper;
+
+    @Autowired
+    public OrderController(OrderService orderService,
+                           ShoppingCartService shoppingCartService,
+                           UserService userService,
+                           OrderDtoMapper orderDtoMapper) {
+        this.orderService = orderService;
+        this.shoppingCartService = shoppingCartService;
+        this.userService = userService;
+        this.orderDtoMapper = orderDtoMapper;
+    }
+
+    @PostMapping("/complete")
+    public void completeOrder(@RequestParam("userId") Long userId) {
```

Comment:
```suggestion
    public void completeOrder(@RequestParam Long userId) {
```
it is redundant if you pass the identically named param

Reply:
done